### PR TITLE
chore(edithomepage): shift constants out

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
There were many constants whose reference wasn't clear in `EditHomepage`. This PR shifts them out for clarity and exposes a helper method to construct the error message.

## Solution
1. shift constants out to `ts` file. 
2. expose util method for getting error object.

### CODE REVIEW

File: src/layouts/EditHomepage/constants.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export const DROPDOWN_ELEMENT_SECTION = {
+  title: "Hero Dropdown Element Title",
+  url: "", // No default value so that no broken link is created
+} as const
+
+export const DROPDOWN_SECTION = {
+  title: "Hero Dropdown Title",
+  options: [],
+} as const
```
Comment (harishv7) [inline src/layouts/EditHomepage/constants.ts:41]: do we need the as const?

File: src/layouts/EditHomepage/utils.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import _ from "lodash"
+
+export const getErrorValues = (
+  obj: Record<string, string>
+): Record<string, string> => {
+  return _.mapValues(obj, () => "")
```
Comment (harishv7) [inline src/layouts/EditHomepage/utils.ts:6]: for err val we just return empty string for all keys?

Comment (seaerchin) [reply src/layouts/EditHomepage/utils.ts:6]: apparently so - see [here](https://github.com/isomerpages/isomercms-frontend/blob/develop/src/layouts/EditHomepage.jsx#L53)

File: src/layouts/EditHomepage/constants.ts (hunk 1 above)
Comment (seaerchin) [reply src/layouts/EditHomepage/constants.ts:41]: yeap - the `as const` is to tell `ts` that this is a `readonly const` value. there's 2 differences with this cast

1. type inference becomes alot stricter - w/ the cast, `DROPDOWN_SECTION.title` is always inferred as `Hero dropdown title` and not `string` 
2. no reassignment of properties - with the cast, we cannot do `DROPDOWN_SECTION.title = "mutated"`; without the cast, we **can**. the `const DROPDOWN_SECTION` applies only to the _identifier_, which means that we can't do `const DROPDOWN_SECTION = ... // something with same shape` but can still do reassignment of properties

Comment (harishv7) [reply src/layouts/EditHomepage/constants.ts:41]: cool, makes sense

**Review by harishv7 — APPROVED** [review]

File: src/layouts/EditHomepage/constants.ts (hunk 1 above)
Comment (QiluXie) [reply src/layouts/EditHomepage/constants.ts:41]: @seaerchin good explanation, using `as const` enforces literal types and read-only properties

**Review by QiluXie — APPROVED** [review]
lgtm

**Comment by seaerchin** [comment]
**not merging this first because this involves a file rename, which is annoying to rebase**

**Pull request merged**